### PR TITLE
pyopae: add device_id property to Python API

### PR DESCRIPTION
**Pull request by r-rojo**
Also reword some of the doc strings

### CODE REVIEW

File: pyopae/pyproperties.cpp [hunk 1]
```diff
@@ -363,6 +367,23 @@ void properties_set_vendor_id(properties::ptr_t props, uint32_t vendor_id) {
   props->vendor_id = vendor_id;
 }
 
+// device id
+const char *properties_doc_device_id() {
+  return R"opaedoc(
+    Get or set the device ID  property of a resource.
+    The device ID is part of the PCI ID and is assibned by the
```
Comment (tswhison) [inline pyopae/pyproperties.cpp]: assigned

Comment (r-rojo) [reply pyopae/pyproperties.cpp]: doh!
Good catch!

File: pyopae/pyproperties.cpp [hunk 2]
```diff
@@ -63,31 +63,33 @@ const char *properties_doc_get() {
...
 
-      vendor_id: The vendor ID in PCI config space.
+      vendor_id (uint16_t): The vendor ID in PCI config space.
+
+      device_id (uint16_t): The device ID in PCI config space.
 
       model (str): The model of the FPGA.
 
-      local_memory_size: The size (in bytes) of the FPGA local memory.
+      local_memory_size (uint64_t): The size (in bytes) of the FPGA local memory.
```
Comment (tswhison) [inline pyopae/pyproperties.cpp]: Do we support this property currently?  Last I remembered, it was a TODO item.

Comment (r-rojo) [reply pyopae/pyproperties.cpp]: It looks like it's still not supported 😞 
In this case, the C++ backing code will throw a `not_supported` exception which will get converted into a Python RuntimeError with the message generated by the `not_supported` exception.
I think this is fine for now but we should probably look at either defining analogous exception classes in Python or convert the C++ exceptions to standard Python exceptions that make sense for the situation.

**Pull request merged**